network/p2p/p2p_sync.go

```diff
@@ -300,3 +288,42 @@ func (n *p2pNetwork) peersWithProtocolsFilter(protocols ...libp2p_protocol.ID) f
 func allPeersFilter(id peer.ID) bool {
 	return true
 }
+
+func waitSubsetOfPeers(
+	logger *zap.Logger,
+	getSubsetOfPeers func(logger *zap.Logger, vpk spectypes.ValidatorPK, maxPeers int, filter func(peer.ID) bool) (peers []peer.ID, err error),
+	vpk spectypes.ValidatorPK,
+	minPeers, maxPeers int,
+	timeout time.Duration,
+	filter func(peer.ID) bool,
+) ([]peer.ID, error) {
+	if minPeers > maxPeers {
+		return nil, fmt.Errorf("minPeers should not be greater than maxPeers")
+	}
+	if minPeers < 0 || maxPeers < 0 {
+		return nil, fmt.Errorf("minPeers and maxPeers should not be negative")
+	}
+	if timeout <= 0 {
+		return nil, fmt.Errorf("timeout should be positive")
+	}
+
+	// Wait for minPeers with a deadline.
+	deadline := time.Now().Add(timeout)
```

Comment:
could've been done with context.Deadline as well. though I don't see a real issue with current approach. once we'll want to introduce shutdown we'll need to pass a context down and combine it with this.

Reply:
this func doesn't accept context (like all other funcs in this struct)

what needs to be done is a huge refactor to every blocking func/method in SSV to accept context and pass it down correctly

though, for graceful shutdown the only important piece is the database (imo), so we can make that refactor a bit smaller